.github/workflows/auto_approve_dependency_PRs.yml

```diff
@@ -0,0 +1,37 @@
+name: Auto Approve Dependency PRs
+on:
+  schedule:
+    - cron: '*/30 * * * *'
+  workflow_dispatch:
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Find dependency PRs
+        id: find_prs
+        run: |
+          gh auth status
+          rm -f dep_PRs_waiting_approval.json
+          gh pr list --repo "${{ github.repository }}" --author "machineFL" --base main --state open --search "status:success review:required" --limit 1 --json number > dep_PRs_waiting_approval.json
+          cat dep_PRs_waiting_approval.json
+          dep_pull_request=$(cat dep_PRs_waiting_approval.json | grep -Eo "[0-9]*")
+          echo $dep_pull_request
+          echo ::set-output name=dep_pull_request::${dep_pull_request}
+        env:
+          GITHUB_TOKEN: ${{ secrets.AUTO_APPROVE_TOKEN }}
+      - name: Approve dependency PRs and enable auto-merge
+        id: approve_prs
+        if: >-
+          ${{
+            steps.find_prs.outputs.dep_pull_request != '' &&
+            steps.find_prs.outputs.dep_pull_request != null &&
+            steps.find_prs.outputs.dep_pull_request > 1
+          }}
+        run: |
+          echo ${{ steps.find_prs.outputs.dep_pull_request }}
+          gh pr review --repo "${{ github.repository }}" --comment --body "auto approve" ${{ steps.find_prs.outputs.dep_pull_request }}
+          gh pr review --repo "${{ github.repository }}" --approve ${{ steps.find_prs.outputs.dep_pull_request }}
+          gh pr merge --repo "${{ github.repository }}" --auto --squash ${{ steps.find_prs.outputs.dep_pull_request }}
```

Comment:
So this will error out if any of the CI checks are red? What if some are still running? 

If this doesn't error out for pending checks, should we change the cron to run every 5 minutes? That way this gets merged as soon as CI finishes + passes.

Reply:
The `status:success review:required` means that it will only auto approve if all CI passes.

I can change it to run every 5 minutes.

Reply:
Will this auto-approve and merge if a reviewer approves but leaves comments on nit-picks/suggestions?

Reply:
This is only for dependency update PRs, such as this:
- https://github.com/alteryx/evalml/pull/3180

Reply:
My hope would be that EvalML reviewers are no longer touching those types of PRs (when they pass all CI checks).